lib/hyacinth/adapters/external_identifier_adapter/datacite/rest_api/v2/api.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+require 'json'
+class Hyacinth::Adapters::ExternalIdentifierAdapter::Datacite::RestApi::V2::Api
+  MINIMUM_REQUIRED_ATTRIBUTES_FOR_FINDABLE_DOI = [:title,
+                                                  :creators,
+                                                  :doi_url,
+                                                  :publisher,
+                                                  :publication_year,
+                                                  :resource_type,
+                                                  :doi_prefix].freeze
+  # Events used when minting a DOI, depends on desired state of minted DOI
+  # see https://support.datacite.org/docs/how-do-i-make-a-findable-doi-with-the-rest-api
+  DOI_STATES = [:draft, :findable, :registered].freeze
+  DOI_MINT_EVENT = { draft: '',
+                     findable: 'publish',
+                     registered: 'hide' }.freeze
+
+  attr_accessor :most_recent_request_body,
+                :most_recent_response
+
+  def initialize(datacite_rest_api_url = DATACITE[:rest_api],
+                 basic_auth_user = DATACITE[:user],
+                 basic_auth_password = DATACITE[:password])
+    @datacite_rest_api_url = datacite_rest_api_url
+    @basic_auth_user = basic_auth_user
+    @basic_auth_password = basic_auth_password
+  end
+
+  # see https://support.datacite.org/reference/dois-2#get_dois-id
+  # 'Returns a doi.'
+  def get_dois(doi)
+    conn = Faraday.new(@datacite_rest_api_url)
+    conn.basic_auth(@basic_auth_user, @basic_auth_password)
+    response = conn.get("/dois/#{doi}")
+    log("API response body: #{response.body}", DATACITE[:log_level]) if DATACITE[:log_api]
+    response
+  end
+
+  def log(log_msg, log_level)
+    return if ['debug', 'info', 'warn', 'error', 'fatal', 'unknown'].exclude? log_level
+    Rails.logger.public_send(log_level, log_msg)
+  end
```

Comment:
I'm curious why you implemented your own logging subsystem in here rather than relying on direct use of the built-in Rails logging system.  It has a limited scope so it won't do any harm, but it may be a little inconvenient to have to control one very specific Hyacinth subsystem with its own logging setting.  A while back, Ben and I had to deal with a library in a Rails app that had its own logging setting rather than inheriting the Rails default, and it made it harder to follow the flow of debug information while we were troubleshooting a problem.

Reply:
To be fix in logger PR